### PR TITLE
[Idea] [Breaking changes!] Add shorthand flags

### PR DESCRIPTION
**Pull request by xmudrii**
I'm starting this as idea to make shorthand flags work.
Current status when you execute `doctl compute droplet delete`:
```
Delete droplet by id or name

Usage:
  doctl compute droplet delete ID [ID|Name ...] [flags]

Aliases:
  delete, d, del, rm


Flags:
      --force   Force droplet delete
  -h, --help    help for delete

Global Flags:
  -t, --access-token string   API V2 Access Token
  -c, --config string         config file (default is $HOME/.config/doctl/config.yaml)
  -o, --output string         output format [text|json] (default "text")
      --trace                 trace api access
  -v, --verbose               verbose output

Error: (droplet.delete) command is missing required arguments
```

If you want to force delete, you'll have to type `--force`. I don't see that natural in Unix world, as it's mostly normal to have shorthand flag (`-f`).

When I was implementing delete confirm in #148, there was no way to do it, and at that point I had no time to play around it.

That's not only for force, it's for all flags.

This is ready code, it's fully usable (I'm not sure about tests tho), and it's working.
I added one more argument for shorthand `Add*Flag` and changed from `*` to `*P` (e.g. `String` to `StringP`).

I feel like it could be done better, but let's discuss and give me idea if you agree with adding shorthand flags.

When you execute same command after chagnes:
```
Delete droplet by id or name

Usage:
  doctl compute droplet delete ID [ID|Name ...] [flags]

Aliases:
  delete, d, del, rm


Flags:
  -f, --force   Force droplet delete
  -h, --help    help for delete

Global Flags:
  -t, --access-token string   API V2 Access Token
  -c, --config string         config file (default is $HOME/.config/doctl/config.yaml)
  -o, --output string         output format [text|json] (default "text")
      --trace                 trace api access
  -v, --verbose               verbose output

Error: (droplet.delete) command is missing required arguments
```
Both `-f` and `--force` are working!

I hope you like idea.

cc @aybabtme @phillbaker @andrewsomething @nanzhong @mchitten

### CODE REVIEW

**Comment by nanzhong** [comment]
Nice, I like it! @aybabtme what do you think?

File: commands/domains.go [hunk 1]
```diff
@@ -38,8 +38,8 @@ func Domain() *Command {
 
 	cmdDomainCreate := CmdBuilder(cmd, RunDomainCreate, "create <domain>", "create domain", Writer,
 		aliasOpt("c"), displayerType(&domain{}), docCategories("domain"))
-	AddStringFlag(cmdDomainCreate, doctl.ArgIPAddress, "", "IP address", requiredOpt())
-
+	AddStringFlag(cmdDomainCreate, doctl.ArgIPAddress, "", "", "IP address", requiredOpt())
+
```
Comment (nanzhong) [inline commands/domains.go]: Tiny nitpick, extra whitespace.

Comment (xmudrii) [inline commands/domains.go]: @nanzhong fixed. :D I hope it's correct now. Also fixed indentation for `volumes.go` at one point.
And yea, commit should be formatting but whatever. :P

File: args_short.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package doctl
+
+const (
+	/**
```
Comment (aybabtme) [inline args_short.go]: I'm curious why this large block of commented out code?

Comment (nanzhong) [inline args_short.go]: My assumption was some placeholder in case we want to introduce more short flags for other things?

Comment (xmudrii) [inline args_short.go]: @nanzhong @aybabtme I left it for placeholder, but I removed it. For now I defined it for force action, but we can add more actions later.

**Comment by nanzhong** [comment]
Thanks @xmudrii for all your contributions. We realize that we've been a little slow to respond, but hoping to change that in the coming year. :) 🎆

**Comment by xmudrii** [comment]
No problem, glad to help. :) And that's not a problem. :)

**Pull request merged**